pkg/services/sqlstore/migrations/user_mig.go

```diff
@@ -140,6 +140,19 @@ func addUserMigrations(mg *Migrator) {
 			SQLite(migSQLITEisServiceAccountNullable).
 			Postgres("ALTER TABLE `user` ALTER COLUMN is_service_account DROP NOT NULL;").
 			Mysql("ALTER TABLE user MODIFY is_service_account BOOLEAN DEFAULT 0;"))
+
+	mg.AddMigration("Add uid column to user", NewAddColumnMigration(userV2, &Column{
+		Name: "uid", Type: DB_NVarchar, Length: 40, Nullable: false, Default: "",
+	}))
+
+	mg.AddMigration("Update uid column values for users", NewRawSQLMigration("").
+		SQLite("UPDATE user SET uid=printf('i%09d',id) WHERE uid IS NULL;").
+		Postgres("UPDATE `user` SET uid='i' || lpad('' || id::text,9,'0') WHERE uid IS NULL;").
+		Mysql("UPDATE user SET uid=concat('i',lpad(id,9,'0')) WHERE uid IS NULL;"))
+
+	mg.AddMigration("Add unique index user_org_id_uid", NewAddIndexMigration(userV2, &Index{
+		Cols: []string{"org_id", "uid"}, Type: UniqueIndex,
```

Comment:
I think the unique index needs to be on only uid. I am pretty sure `org_id` in user table is just used to determine last used org so this value will change. 

What this will do is prevent two users with the same uid to be active in the same org at once

Reply:
> is just used to determine last used org

🤯 good to know!  so much mystery in the orgs